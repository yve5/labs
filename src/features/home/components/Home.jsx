import React from 'react';
import { Link } from 'react-router-dom';
import { HOME_LINKS } from '../resources/constants';

const Home = () => (
  <div className="container">
    <h1>Labs</h1>

    <ul>
      {HOME_LINKS.map(({ href, inner, label }, index) => (
        <li key={index}>
          {inner ? (
            <Link to={href}>{label}</Link>
          ) : (
            <a href={href} target="blank">
              {label}
            </a>
          )}
        </li>
      ))}
    </ul>
  </div>
);

export default Home;
