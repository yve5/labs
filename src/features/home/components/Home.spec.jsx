import { create } from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';
import Home from './Home';

describe('Home', () => {
  it('should match expected snapshot. default', () => {
    const component = create(
      <HashRouter>
        <Home />
      </HashRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
