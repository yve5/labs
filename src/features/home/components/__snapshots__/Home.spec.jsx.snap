// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match expected snapshot. default 1`] = `
<div
  className="container"
>
  <h1>
    Labs
  </h1>
  <ul>
    <li>
      <a
        href="https://yve5.github.io/pairs"
        target="blank"
      >
        Pairs Game
      </a>
    </li>
    <li>
      <a
        href="https://yve5.github.io/lipstickcss"
        target="blank"
      >
        Lipstick CSS
      </a>
    </li>
    <li>
      <a
        href="https://yve5.github.io/satisfaction"
        target="blank"
      >
        Confidence
      </a>
    </li>
    <li>
      <a
        href="#/more"
        onClick={[Function]}
      >
        More...
      </a>
    </li>
  </ul>
</div>
`;
